gateway-ha/src/main/java/io/trino/gateway/ha/router/QueryCountBasedRouter.java

```diff
@@ -224,7 +224,11 @@ private synchronized Optional<String> getBackendForRoutingGroup(String routingGr
     @Override
     public String provideAdhocBackend(String user)
     {
-        return getBackendForRoutingGroup("adhoc", user).orElseThrow();
+        return getBackendForRoutingGroup("adhoc", user).orElseThrow(() -> {
+            RouterException adhocRoutingGroupException = new RouterException("did not find any cluster for the adhoc routing group");
+            log.error(adhocRoutingGroupException.getMessage(), adhocRoutingGroupException);
```

Comment:
Update the logger to `io.airlift.log.Logger` and
```suggestion
            log.error(adhocRoutingGroupException, adhocRoutingGroupException.getMessage())
```
This is the last class using the slf4j logger, so you can remove that dependency after the change